src/Microsoft.Azure.WebJobs.Extensions.Storage/Blobs/Listeners/BlobTriggerExecutor.cs

```diff
@@ -49,18 +62,22 @@ public async Task<FunctionResult> ExecuteAsync(ICloudBlob value, CancellationTok
 
             if (possibleETag == null)
             {
+                Logger.BlobHasNoETag(_logger, _functionDescriptor.LogName, value.Name, pollId, triggerSource);
+
                 // If the blob doesn't exist and have an ETag, don't trigger on it.
                 return new FunctionResult(true);
             }
 
-            var receiptBlob = _receiptManager.CreateReference(_hostId, _functionId, value.Container.Name,
+            var receiptBlob = _receiptManager.CreateReference(_hostId, _functionDescriptor.Id, value.Container.Name,
                 value.Name, possibleETag);
 
             // Check for the completed receipt. If it's already there, noop.
             BlobReceipt unleasedReceipt = await _receiptManager.TryReadAsync(receiptBlob, cancellationToken);
 
             if (unleasedReceipt != null && unleasedReceipt.IsCompleted)
             {
+                Logger.BlobAlreadyProcessed(_logger, _functionDescriptor.LogName, value.Name, possibleETag, pollId, triggerSource);
```

Comment:
Does this mean that on every container scan, we'll be logging a line for all blobs that are already processed?